cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -2497,3 +2498,16 @@ class AdditionalCallTarget extends Unit {
    */
   abstract Declaration viableTarget(Call call);
 }
+
+/**
+ * Gets a function that may be called by `call`.
+ *
+ * Note that `call` may be a call to a function pointer expression.
+ */
+Function getARuntimeTarget(Call call) {
```

Comment:
I'm always slightly confused. Is `DataFlowUtil` the file whose contents users may still depend upon, although it's located in `internal`?

Reply:
Correct: https://github.com/github/codeql/blob/7d62cdad71eeeaa9c79880f005d4ea00718de32e/cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowImplSpecific.qll#L14

I don't know why this was done like this. This file structure is also present in the old AST dataflow, and we just copied it for IR dataflow.

We could actually change this, and maybe rename `DataFlowUtil` to `DataFlowPublic` to clear this confusion like it's been done for newer languages:

<img width="539" alt="Screenshot 2024-10-17 at 11 01 15" src="https://github.com/user-attachments/assets/91bee586-0f71-4c7e-b8b0-fbb6910fa16f">

(although it looks like every language still stores the file in `internal` for some reason 🤔)